td-shim/src/secure_boot.rs

```diff
@@ -171,8 +171,9 @@ impl<'a> PayloadVerifier<'a> {
                 signature = &signed_payload[offset..offset + 96];
 
                 // Uncompressed public key
-                formated_public_key.push(0x04);
-                formated_public_key.extend_from_slice(public_key);
+                formated_public_key[0] = 0x04;
+                formated_public_key[1..1 + public_key.len()].copy_from_slice(public_key);
+                public_key_size = 97;
```

Comment:
why hardcode 97?

Reply:
For RSA, the public key input is encoded in DER format.

See https://github.com/briansmith/ring/blob/main/src/rsa.rs#L44-L54

For ECDSA, the public key input is one encoding byte (indicating compression) and two 48 bytes point data:

See https://github.com/briansmith/ring/blob/main/src/ec/suite_b/public_key.rs#L29-L48 and https://github.com/briansmith/ring/blob/main/src/ec/suite_b/ops.rs#L224-L245